frontend/catalyst/__init__.py

```diff
@@ -88,6 +87,7 @@
 from catalyst.autograph import *
 from catalyst.autograph import __all__ as _autograph_functions
 from catalyst.compiler import CompileOptions
+from catalyst.jax_tracer import dynamic_one_shot
```

Comment:
Generally we only import names into the top-level package that we want to make part of the "public api". In this case, I don't think the plan is for anyone to use the transform manually, is it?

Reply:
With the `mcm_method="dynamic-one-shot"` configuration being the official way to select the execution mode I don't think there is a need to export the transform. @josh146 any thoughts?

Reply:
My vote is to import it top level, because while the new suggested behaviour is to use `mcm_method="one-shot"`, users can still use the transform directly, and it is still exposed top level in pennylane.

Reply:
Does it really make sense to have two ways to do the same thing though? Personally I cannot see why.

Doesn't need to hold up this PR but something to consider in PL as well.

Reply:
I am not too opinionated here, but would be okay having it accessible to users (whether that means imported top level, or available in the docs in a submodule).

In the past with PL, we have had cases where we have been helping users, and noticed cases where using the transform directly is useful. E.g.:

- In almost all of our documentation, we specify users to use `qnode(dev, diff_method="param_shift")` etc.

- However, having `qml.gradients.param_shift` is useful if:

  - the user wants to view/draw/inspect the transformed circuits (can happen somewhat frequently)
  - the user wants to _train_ the transform (e.g., the shift size). Doesn't happen often.

Reply:
Note that this transform works differently than regular PL transforms, it takes a QNode and produces a classical function. If inspecting the circuit is a goal this would probably not be the way to do it. There also aren't any parameters to train.

Imo this is an internal function implementing one of the quantum execution modes. 

But if given the above you still think it should be exposed publicly I'm okay with that :)